src/entrypoints/recorder.ts

```diff
@@ -309,7 +309,7 @@ function initXhrObserver(cb: networkCallback, win: IWindow, options: Required<Ne
                         .then((entry) => {
                             const requests = prepareRequest({
                                 entry,
-                                method: req.method,
+                                method: method,
```

Comment:
this is the fix... the method is provided to the XHR function when it is called so we should use it